sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorReceiver.java

```diff
@@ -183,8 +186,16 @@ protected Message decodeDelivery(Delivery delivery) {
 
         final Message message = Proton.message();
         message.decode(buffer, 0, read);
-
-        delivery.settle();
-        return message;
+        ExtendedAmqpMessage extendedAmqpMessage = null;
+        if (receiver.getSenderSettleMode() == SenderSettleMode.SETTLED) {
+            // No op. Delivery comes settled from the sender
+            delivery.disposition(Accepted.getInstance());
+            delivery.settle();
+            extendedAmqpMessage =  new ExtendedAmqpMessage(message, null);
+        } else {
+            extendedAmqpMessage =  new ExtendedAmqpMessage(message,
+                ByteBuffer.wrap(delivery.getTag()));
+        }
+        return extendedAmqpMessage;
```

Comment:
Do we need to maintaining type information upon return from this method? Currently, once this method returns we will lose the concrete type and the caller will need to check the object type if they want to access the extended message properties.